rust/kcl-lib/src/std/array.rs

```diff
@@ -19,9 +20,11 @@ use crate::{
 pub async fn map(exec_state: &mut ExecState, args: Args) -> Result<KclValue, KclError> {
     let array: Vec<KclValue> = args.get_unlabeled_kw_arg("array")?;
     let f: &FunctionSource = args.get_kw_arg("f")?;
-    let meta = vec![args.source_range.into()];
     let new_array = inner_map(array, f, exec_state, &args).await?;
-    Ok(KclValue::MixedArray { value: new_array, meta })
+    Ok(KclValue::HomArray {
+        value: new_array,
+        ty: RuntimeType::any(),
```

Comment:
I thought it would be nice to use the return type of `f` here since we already coerce to it when it's present on user-defined functions. For stdlib functions, do we have that information? I don't think we have it for functions implemented in Rust. I think it would be a breaking change to try to add this later.